app/routes/__auth/login.tsx

```diff
@@ -64,7 +64,7 @@ export default function LoginRoute() {
 export const action = async ({ request }) => {
   const formData = await request.formData();
   const errors = await validate(formData, loginValidators);
-  if (errors) return { errors };
+  if (errors) return { errors, formData: Object.fromEntries(formData) };
```

Comment:
return the submitted values as part of actionData so they can be used to hydrate the form inputs. this way the form won't feel liek it reset on error if no JS.